### ISSUE TITLE
Fan is "dumping" in "low-speed fan" profile

### ISSUE THREAD

**Issue by erazortt** [cronosun/atrofac#5]
I have not to been able to figure out why, but using the "low-speed fan" profile for me the fan is always pumping (on-off in 4 sec intervalls), independently of the tempereture. This is the curve I'm using:

  - name: Silent (low-speed fan)
    plan: silent
    refresh_interval_sec: ~
    cpu_curve: "20c:0%,45c:5%,55c:10%,60c:15%,70c:31%,80c:49%,90c:56%,100c:56%"
    gpu_curve: "20c:0%,50c:5%,55c:10%,60c:15%,70c:34%,80c:51%,90c:61%,100c:61%"

**Comment by Yarn** [comment]
I played with this a bit in armoury crate and it seems to happen some of the time. 

Based on some very limited testing it seems like this issue happens if you have it set to a manual profile on boot. If I set it to silent, rebooted, then set it to manual, it seemed to work. If it was manual on boot it would fluctuate on manual but not on other settings (switching back and forth doesn't seem to fix it).

My guess is that the driver/acpi isn't initializing something on the EC chip when it's set to manual.

I'm working on getting fan curves working in linux and seeing the same fluctuating fan speed issue. (Well it stopped happening and I'm not sure why, if it comes back after booting into windows I can look into it more.)

**Comment by cronosun** [comment]
>  Does somebody see the same behaviour?

Not really, definitely not in the "Fanless" profile (in the "low-speed fan"-profile sometimes; but this got better with the new fan curve from the latest atrofac version).

Possible causes:

- Do you have a different G14 model? (mine is the GA401IU with bios 212).
- Do you use the latest version of atrofac with the new fan curves? (however, the latest version should not change anything in regard to the fanless-profile). Release 2020-07-07.
- Nvidia GPU is disabled?
- How much power does your PC drain? See Step 4 in the doc (https://github.com/cronosun/atrofac).

**Note**:

> cpu_curve: "20c:0%,45c:5%,55c:10%,60c:15%,70c:31%,80c:49%,90c:56%,100c:56%"
> gpu_curve: "20c:0%,50c:5%,55c:10%,60c:15%,70c:34%,80c:51%,90c:61%,100c:61%"

Those curves seem to be outside the allowed limits ("20c" should be within 30c-39c). See: https://github.com/cronosun/atrofac/blob/master/ADVANCED.md

**Note 2**:

When I tried to solve the problem with pumping fan speeds in the "low-speed fan"-profile (see latest release) I had to fiddle  quite a bit with the gpu/gpu fan curves. I ended up with those values:

- CPU: 30c:10%,49c:10%,59c:10%,69c:10%,79c:31%,89c:49%,99c:56%,109c:56%
- GPU: 30c:0%,49c:0%,59c:0%,69c:0%,79c:34%,89c:51%,99c:61%,109c:61%

I could not go below 10% (going below 8-10% causes the fan to pump) - higher values like 12% work even better... Starting temperature had to be 30c. I had to chose a constant percentage between 30c to 78c... Depending on your G14 model, environmental temperature, ... you might need to tweak those values a bit.

**Comment by erazortt** [comment]
In the meantime I also think this has something to do how the laptop is booted. It appeared that when it was hot and restarted it would go into pumping. But when it was booting from cold it was normal. But that really does not make any sense so thats probably bs.

- I have the IV with 212 bios.
- I had the 2020-05-26 release.
- As long as I'm not on the exernal display, and I'm not doing 3D stuff the 2060 is in complete sleep mode (all 0's in hwinfo). And the Nvidia activity monitor shows no program being run using it.
- It drains around 6.5W in total when on battery (the smallest I have seen was 5.5W).

So when it was pumping for me, it never stopped, independently of the temperature and of the fan speed. It would pump up to 70°C (haven't tried higher).

The 20c point was a try to see if it can be due to this, since I though at the boot the temp might be below 30c and than, perhaps since it does not have anything assigned, I though it might be going into a wierd undefined state which would be the cause of thiis pumping.

**Comment by Yarn** [comment]
I have the IV with 212 bios. Given the issue persists in linux when I'm essentially writing directly to the EC and seems to either happen or not happen on a given boot it certainly seems to be either some initial conditions on boot or some state the ec is carrying over between boots. I'll jump back in windows later and see if the pattern I saw before is still happening.

**Comment by Yarn** [comment]
Additional observation, the issue does seem to show up when it was working at boot. A reboot doesn't seem to fix it. No idea what causes it to start/stop happening at this point.

**Comment by killthrash** [comment]
I was having the same issue. This is what worked for me -

1. Uninstalled MyAsus
2. Turn keyboard backlight completely off and shut down
3. Boot

For some reason, if I reboot with the keyboard backlight on, the fans oscillate back and forth from 2k-3k rpm.

**Comment by Yarn** [comment]
It looks like it can start/stop going into and coming out of sleep. Going into sleep a few times until it stops is at least a not too annoying way to fix the issue.

**Comment by levaid** [comment]
After hours of troubleshooting yesterday, the fan curves started working well and not ramping up, but today, after waking up from hibernate, fans were pulsing. I am also using the IV model with R9 4900HS and RTX2060.

**Comment by digioncz** [comment]
Hi, I have GA401II (Ryzen 4600HS, 1650TI) and duming fan are connected with charging. While Zephyrus go on battery then everything worked. When plug-in charger then fan is jumps between 2-3k while setting is 2300 RPM.

**Comment by arthurprs** [comment]
Did anyone figure this out? I got a nice curve that worked for me, then in the evening (reboot) the same curve would just pump the fans on/off...

**Comment by killthrash** [comment]
> I was having the same issue. This is what worked for me -
> 
> 1. Uninstalled MyAsus
> 2. Turn keyboard backlight completely off and shut down
> 3. Boot
> 
> For some reason, if I reboot with the keyboard backlight on, the fans oscillate back and forth from 2k-3k rpm.



> I was having the same issue. This is what worked for me -
> 
> 1. Uninstalled MyAsus
> 2. Turn keyboard backlight completely off and shut down
> 3. Boot
> 
> For some reason, if I reboot with the keyboard backlight on, the fans oscillate back and forth from 2k-3k rpm.

Update:

In some cases, when I boot with my USB-C dock plugged in, the fans pump up and down. When I unplug the dock and boot, the utility works as designed. Then I just plug the dock in after I'm in Windows.

**Comment by kapythone** [comment]
I found an interesting post saying this might have something to do with BIOS protection mechanism (https://www.reddit.com/r/ZephyrusG14/comments/kumf87/asus_zephyrus_g14_fan_issue/):


> This issue has to do with the BIOS. When the CPU temps are below 30°C, the bios triggers and emergency mode. When this happens, the CPU starts to throttle and the fans rev to max. This is meant to happen when the PC gets too hot.


I also noticed the fan ramping up and down happens only when I set a low-value fan curve (1%) and during a cold boot-up or wake-up.

**Comment by albza** [comment]
> It looks like it can start/stop going into and coming out of sleep. Going into sleep a few times until it stops is at least a not too annoying way to fix the issue.

solved with this

**Comment by kaizenblumenblupper** [comment]
Can't fix it. Hybernation causes fans to flactuate up and down after I wake the laptop. Seems sleep doesn't do the same thing weirdly. 
Did anybody try to identify the exact cause? Might it be the vendor software?

**Comment by rumcajs997** [comment]
Same issue here, G14 2021 model with 3050Ti and 120hz display. It was working perfectly fine for few weeks and suddenly started to act weirdly over the last couple of days. It turns on the fan once it hit 49C (which is fine as I set 49C - 10%) but turns it off when the temp. drops barely 2C after few seconds, then it hits 49C after few seconds again and turns on the fan and so on.

One thing that I changed over those few days is that I plugged in an external monitor over USB-C and kept the laptop plugged in. Which means that it might be some conflict with either nVidia graphic card (which is on while using USB-C monitor) or something with keeping the laptop plugged in
